BloomFilter/Program.cs

```diff
@@ -0,0 +1,346 @@
+﻿// ============================================================================
+// FileName: Program.cs
+//
+// Description:
+// A minimal BitCoin client that searches for all transactions related to a single
+// address using a bloom filter. The steps are:
+//
+// 1. Load blockchain headers from disk (if not available will be requested from 
+//    the full node but that takes longer),
+// 2. Connect to a full BitCoin node (this samle is hard coded to use the loopback 
+//    address so the full node will need to be on the same machine),
+// 3. Keep the blockchain headers synchronised with the full node and periodically
+//    save them to disk,
+// 4. Once the blockchain headers are synchronised use set a bloom filter and get 
+//    all blocks within a certain date range to check for relevant transactions.
+//
+// NOTE: This sample does not do any block verification and is NOT suitable for 
+// any kind of use on the main BitCoin network.
+//
+// Dependencies:
+// The program relies on NBitCoin (https://github.com/MetacoSA/NBitcoin) for the 
+// underlying BitCoin primitives.
+//
+// Hints:
+// The original prupose for this sample was to gain an understanding of the BitCoin
+// protocol. An invaluable tool for anyone attempting the same thing is WireShark
+// (https://www.wireshark.org/) which has a builtin BitCoin protocol decoder. To
+// use WireShark with the loopback adapter on Windows install Npcap (https://nmap.org/npcap/).
+//
+// The command line used for the local bitcoin full node:
+// "C:\Program Files\Bitcoin\daemon\bitcoind" -printtoconsole -datadir=f:\temp\bitcoind -server -testnet -debug=1 -bind=[::1]:18333
+//
```

Comment:
Actually your code should work just fine with `"C:\Program Files\Bitcoin\daemon\bitcoind" -testnet`, but yeah your parameters are better to understand what is going on.